controller/server/info.ml

```diff
@@ -46,9 +46,26 @@ let of_file f =
   |> Mustache.of_string
   |> return
 
+(** Format a 32-character string as groups of 4; leave other strings unchanged. *)
+let format_machine_id s =
+  if String.length s == 32 then
+    String.concat
+      "-"
+      [ String.sub s 0 4
+      ; String.sub s 4 4
+      ; String.sub s 8 4
+      ; String.sub s 12 4
+      ; String.sub s 16 4
+      ; String.sub s 20 4
+      ; String.sub s 24 4
+      ; String.sub s 28 4
+      ]
```

Comment:
While this works, it could be shorten using:

```ocaml
List.init 8 (fun x -> String.sub s (x * 4) 4)
```

Cons is that it may be less evident, but pro is that there is less potential error.

Another idea is to use a dedicated function:

```ocaml
let rec grouped n s =
  let l = String.length s in
  if l == 0 then
    []
  else if l <= n then
    [s]
  else
    List.cons (String.sub s 0 n) (grouped n (String.sub s n (l - n)))
```

The Ocaml std library does not contain a lot of functions. Scala is good on this:

```scala
scala> "1234ABCDZZZZAAAA1234ABCDZZZZAAAA".grouped(4).mkString("-")
res6: String = 1234-ABCD-ZZZZ-AAAA-1234-ABCD-ZZZZ-AAAA
```

Reply:
Yes, I thought about this as well. The OCaml stdlib doesn't provide much and I agree that functions like those offered by Scala would be very useful here.

I am not so sure that introducing such generic code here is a win, however. While I _believe_ your function to be correct, I _know_ my code and your abbreviation (`List.init 8 (fun x -> String.sub s (x * 4) 4)`) to be correct. Our non-generic solutions are also about as short.

I think I'll go with your one-liner, which is probably clearest.